File: parantier-api/src/main/java/com/mapo/palantier/pilot/domain/Pilot.java
package com.mapo.palantier.pilot.domain;

import java.time.LocalDateTime;

public class Pilot {

    private Long id;
    private String title;
    private String content;
    private String topic;
    private String status;
    private String priority;
    private Long reporterId;
    private String reporterName;
    private Long assigneeId;
    private String assigneeName;
    private String dueDate;
    private Integer orderNum;
    private Boolean isArchived;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Pilot() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public String getTopic() { return topic; }
    public void setTopic(String topic) { this.topic = topic; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public String getPriority() { return priority; }
    public void setPriority(String priority) { this.priority = priority; }

    public Long getReporterId() { return reporterId; }
    public void setReporterId(Long reporterId) { this.reporterId = reporterId; }

    public String getReporterName() { return reporterName; }
    public void setReporterName(String reporterName) { this.reporterName = reporterName; }

    public Long getAssigneeId() { return assigneeId; }
    public void setAssigneeId(Long assigneeId) { this.assigneeId = assigneeId; }

    public String getAssigneeName() { return assigneeName; }
    public void setAssigneeName(String assigneeName) { this.assigneeName = assigneeName; }

    public String getDueDate() { return dueDate; }
    public void setDueDate(String dueDate) { this.dueDate = dueDate; }

    public Integer getOrderNum() { return orderNum; }
    public void setOrderNum(Integer orderNum) { this.orderNum = orderNum; }

    public Boolean getIsArchived() { return isArchived; }
    public void setIsArchived(Boolean isArchived) { this.isArchived = isArchived; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}

File: parantier-front/src/entities/pilot/types/pilot.ts
export type PilotStatus = "TODO" | "IN_PROGRESS" | "DONE" | "HOLD";
export type PilotPriority = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";

export interface Pilot {
  id: number;
  title: string;
  content: string;
  topic: string; // 자유 텍스트 (React, Java, Docker 등)
  status: PilotStatus;
  priority: PilotPriority;
  reporterId: number;
  reporterName: string;
  assigneeId?: number;
  assigneeName?: string;
  dueDate?: string;
  orderNum?: number;
  isArchived?: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePilotRequest {
  title: string;
  content: string;
  topic: string;
  status?: PilotStatus;
  priority?: PilotPriority;
  assigneeId?: number | null;
  dueDate?: string | null;
}

export interface UpdatePilotRequest {
  title: string;
  content: string;
  topic: string;
  status?: PilotStatus;
  priority?: PilotPriority;
  assigneeId?: number | null;
  dueDate?: string | null;
}

export interface PilotListResponse {
  items: Pilot[];
  total: number;
  page: number;
  limit: number;
}

export interface PilotFilters {
  topic?: string;
  status?: PilotStatus;
  priority?: PilotPriority;
  assigneeId?: number;
  keyword?: string;
  sortBy?: "created" | "priority" | "status" | "dueDate";
  isArchived?: boolean;
  page?: number;
  limit?: number;
}
